src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -193,9 +193,12 @@ void Graph::Replicate(const std::shared_ptr<const ov::Model> &model,
         auto parentNode = op2node[unusedOutput.get_node_shared_ptr()];
         const auto port = unusedOutput.get_index();
         const auto nodeName = std::string("stub_") + std::to_string(unusedOutput.get_index()) + "_" + parentNode->getName();
+        // WA: avoid PagedAttention's second output reorder.
```

Comment:
We'd better not to hardcode here and should find the place where the output precision is changed to f16.

Reply:
> Any updates here. It has to be resolved before the merge.

still address it. The root cause is `ConvertPrecision` transformation. Need to deal with it carefully and avoid affecting gpu.

Reply:
@dmitry-gorokhov @luo-cheng2021 
remove such wa.
The root cause is that the PagedAttentionExtension op doesn't provide a mechanism to change the 2nd output dtype.
I make some changes.
1. add a `set_out_type` member func to PagedAttentionExtension op.
2. When execute `validate_and_infer_types` in PagedAttentionExtension, it will determine output type. It won't break the GPU path.
3. add a `fuse_type_to_pa` in CPU plugin, which is a extend to `ConvertPrecision`. It is used to specify the correct type for `PagedAttentionExtension` 's 2nd output type.

Reply:
@xczhai Just for my better understanding: could you please descrive the pattern there Reorder is inserted? Like next op after PA expected fp16 on its input?

Reply:
> @xczhai Just for my better understanding: could you please descrive the pattern there Reorder is inserted? Like next op after PA expected fp16 on its input?

Okay.

1. At the very beginning, the PA op spec describes the two outputs type is aligned with input0 type. As a result, PA's outputs type is `f32` when entering CPU plugin.
2. During CPU plugin transformation,  `ConvertPrecision` will convert or fuse the op's type. As a result, PA's two outputs type is `f16`. But remember PA's 2nd output is dangle without any child and `Result` node.
3. In construct graph, all the `dangle` output will be wrapped by `Result` node and the type is aligned with output type. In this case, the specific pattern is `PA's 2nd output` --> `Result(f16)`
4. But in CPU node design, PA's 2nd output is always `f32`. So the pattern is `PA's 2nd output(f32)` --> `Result(f16)`.
5. The following ResolveConflict logic scan this pattern and then insert a `Reorder`. So the pattern becomes `PA's 2nd output(f32)` --> `Reorder` --> `Result(f16)`